plugins/PostProcessingPlugin/scripts/FilamentChange.py

```diff
@@ -74,22 +93,26 @@ def execute(self, data: List[str]):
         later_retract = self.getSettingValueByKey("later_retract")
         x_pos = self.getSettingValueByKey("x_position")
         y_pos = self.getSettingValueByKey("y_position")
+        marlin_conf = self.getSettingValueByKey("marlin")
 
         color_change = "M600"
 
-        if initial_retract is not None and initial_retract > 0.:
-            color_change = color_change + (" E%.2f" % initial_retract)
-
-        if later_retract is not None and later_retract > 0.:
-            color_change = color_change + (" L%.2f" % later_retract)
+        if marlin_conf: 
+            color_change = color_change + " ; Generated by FilamentChange plugin\n"            
+        else:
+            if initial_retract is not None and initial_retract > 0.:
+                color_change = color_change + (" E%.2f" % initial_retract)
 
-        if x_pos is not None:
-            color_change = color_change + (" X%.2f" % x_pos)
+            if later_retract is not None and later_retract > 0.:
+                color_change = color_change + (" L%.2f" % later_retract)
 
-        if y_pos is not None:
-            color_change = color_change + (" Y%.2f" % y_pos)
+            if x_pos is not None:
+                color_change = color_change + (" X%.2f" % x_pos)
 
-        color_change = color_change + " ; Generated by FilamentChange plugin\n"
+            if y_pos is not None:
+                color_change = color_change + (" Y%.2f" % y_pos)
+                
+            color_change = color_change + " ; Generated by FilamentChange plugin\n"
```

Comment:
Might as well just unindent this line and remove the `if` case of the if-else clause. They are (and should stay) the same. And this one has no whitespace at the end of the line.